apps/dcos_overlay/src/dcos_overlay_poller.erl

```diff
@@ -26,8 +27,16 @@
 -endif.
 
 -define(SERVER, ?MODULE).
--define(MIN_POLL_PERIOD, 30000). %% 30 secs
--define(MAX_POLL_PERIOD, 120000). %% 120 secs
+
+%% A failed request is retried sooner than a successful one.
+%% The rationale is, the overlay configuration doesn't change
+%% frequently and the cost of applying such a configuration
+%% far outweighs the cost of fetching it from Mesos over HTTP.
+-define(MIN_SUCCESS_POLL_PERIOD, 30000).  %% 30  secs
+-define(MAX_SUCCESS_POLL_PERIOD, 120000). %% 120 secs
+-define(MIN_FAILURE_POLL_PERIOD, 1000).   %% 1   sec
```

Comment:
I think 1 second is too frequently. To update overlay configuration you should re-configure the whole dc/os cluster and restart all dc/os systemd units. We don't really need backoff here, but if we have it, it shouldn't poll mesos overlay agent module more often than 30 seconds in any cases. One configuration for success and failure case is fine. Here is what I saw once I stopped mesos slave:

```
18:10:11.301 [warning] Overlay Poller could not poll: {failed_connect,[{to_address,{"172.17.0.4",5051}},{inet,[inet],econnrefused}]}
18:10:12.305 [warning] Overlay Poller could not poll: {failed_connect,[{to_address,{"172.17.0.4",5051}},{inet,[inet],econnrefused}]}
18:10:13.357 [warning] Overlay Poller could not poll: {failed_connect,[{to_address,{"172.17.0.4",5051}},{inet,[inet],econnrefused}]}
18:10:15.207 [warning] Overlay Poller could not poll: {failed_connect,[{to_address,{"172.17.0.4",5051}},{inet,[inet],econnrefused}]}
18:10:19.537 [warning] Overlay Poller could not poll: {failed_connect,[{to_address,{"172.17.0.4",5051}},{inet,[inet],econnrefused}]}
```

Reply:
Exactly. This was one of the reasons why I brought it up a few days back in the channel. And there are two options that make sense in my opinion:

1. Not backing off at all. Why backing off up to 2 minutes, and not to `+infinity`, if changes to overlays rarely happen if ever? Especially, if `dcos-net` would be restarted anyways in case of such a change. It would be simpler just to poll the endpoint frequently enough (but not too frequently) and calculate a hash of Mesos response, and apply a configuration if its hash is different.
2. Since there was a exponential backoff in the first place for the success case, it follows that there should be one for the failure case too, and its minimal delay should be smaller, because fetching is cheap. Basically, what's stated in the comment fully applies here.

The way I see it, one configuration is self-contradicting which makes you wonder just like it made me wonder when I read it for the first time.

So what do we choose?

cc @GoelDeepak

Reply:
I vote for option 1

Reply:
it makes sense to poll overlay module on start more often cuz it may take some time to get the configuration from master overlay module. I'd just poll it every 30 seconds with no backoff and  no caching/hashing logic at all.

Reply:
As far as I understand, there was a (performance-related) reason to introduce an exponential backoff retries for successful requests back then. Something has changed that renders unneeded anymore? Are you pro not doing hash calculation in order to periodically ensure that the configuration is actually applied?

Reply:
We don't really care about performance in the poller, it polls a few tiny json objects.

Reply:
> As far as I understand, there was a (performance-related) reason to introduce an exponential backoff retries for successful requests back then. Something has changed that renders unneeded anymore? Are you pro not doing hash calculation in order to periodically ensure that the configuration is actually applied?

There were two concerns when we introduced this functionality back then:
1. we wanted to converge faster in case of failure
2. Reduce unnecessary cpu cycles that in applying the same configuration over and over again. 

We cannot use a hash-based approach because we have seen instances where some other application would change the configuration unhood which would break overlay. So, the trick is to keep applying the config after a fixed interval.

Reply:
Should I revert this file then?

Reply:
yeah, should be fine.